wandb/sdk/launch/sweeps/scheduler.py

```diff
@@ -191,21 +216,23 @@ def _add_to_launch_queue(
             project=self._project,
             entity=self._entity,
             queue=self._launch_queue,
-            resource=resource,
             entry_point=entry_point,
-            # params=params,
+            resource=self._resource,
+            resource_args=self._resource_args,
             run_id=run_id,
         )
-        self._runs[run_id].queued_run = queued_run
-        _msg = f"Added run to Launch RunQueue: {self._launch_queue} RunID:{run_id}."
+        with self._threading_lock:
```

Comment:
`self._runs` is shared amongst the threads so it can change length during execution. The lock prevents modification while one of the threads is using it.